File: csharp/simple-cipher/SimpleCipher.cs
using System;
using System.ComponentModel.DataAnnotations;

public class SimpleCipher
{
    private char[] keys;
     public SimpleCipher()
     {
         Random rand = new Random();
         
         string key = "";
         for (int i = 0; i < 100; i++)
         {
             int randValue = rand.Next(0, 26);
             char letter = Convert.ToChar(randValue + (int)'a');
             key += letter;
         }
         Key = key;
         keys = Key.ToCharArray();
     }

     public SimpleCipher(string key)
     {
         Key = key;
         keys = Key.ToCharArray();
     }

    public string Key { get; private set; }

    string Change(string input, bool encode)
    {
        char[] letters = input.ToCharArray();
        string output = "";
        for (int i = 0; i < input.Length; i++)
        {
            if (encode)
                output += Convert.ToChar((int)letters[i] + (int)keys[i] - (int)'a');
            else
                output += Convert.ToChar((int)letters[i] - ((int)keys[i] - (int)'a')); 
        }

        return output;
    }
    
    public string Encode(string plaintext)
    {
        return Change(plaintext, true);
    }

    public string Decode(string ciphertext)
    {
        return Change(ciphertext, false);
    }
}

File: ruby/simple-cipher/simple_cipher.rb
class SimpleCipher
  @key

  def initialize(key = "")
    @key = key ? key : ("a".."z").to_a.sample(100).join
  end

  def key
    @key
  end

  def change(text, encode)
    keys = @key.chars
    letters = text.chars
    output = ""
    (0...letters.size).each do |i|
      if encode
        ascii = letters[i].to_i + keys[i].to_i - 'a'.to_i
        if ascii > 'z'.to_i
          ascii -= 26
        end
      else
        ascii = letters[i].to_i - keys[i].to_i - 'a'.to_i
        if ascii < 'a'.to_i
          ascii += 26
        end
      end
      output += ascii.chr
    end
    output
  end

  def encode(text)
    change(text, true)
  end

  def decode(text)
    change(text, false)
  end
end
